### PR TITLE
Add creation of manage identity when crating SQL server. Add role assignments as StorageBlobContributor to the new storage

### PR DESCRIPTION
**Pull request by shimonar1171**
Add creation of manage identity when crating SQL server.
Add role assignments as StorageBlobContributor to the new storage

### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use a parameter for resource locations with the defaultValue set to resourceGroup().location
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values)
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

*
*
*

### CODE REVIEW

**Comment by shimonar1171** [comment]
@talhers 
@yoavfr

**Comment by shimonar1171** [comment]
@bmoore-msft

File: 101-sql-logical-server/azuredeploy.json [hunk 1]
```diff
@@ -56,14 +56,19 @@
        "serverResourceGroupName": "[resourceGroup().name]",
        "subscriptionId": "[subscription().subscriptionId]",
        "uniqueStorage": "[uniqueString(variables('subscriptionId'), variables('serverResourceGroupName'), parameters('location'))]",
-       "storageName": "[tolower(concat('sqlva', variables('uniqueStorage')))]"
+       "storageName": "[tolower(concat('sqlva', variables('uniqueStorage')))]",
+       "uniqueRoleGuid": "[guid(uniqueString(variables('subscriptionId'), variables('serverResourceGroupName'), parameters('serverName'), parameters('location')))]",
+       "StorageBlobContributor": "[concat('/subscriptions/', variables('subscriptionId'), '/providers/Microsoft.Authorization/roleDefinitions/', 'ba92f5b4-2d11-453d-a403-e96b0029c9fe')]"
```
Comment (bmoore-msft) [inline 101-sql-logical-server/azuredeploy.json]: remove concat and use subscriptionResourceId()

File: 101-sql-logical-server/azuredeploy.json [hunk 2]
```diff
@@ -56,14 +56,19 @@
        "serverResourceGroupName": "[resourceGroup().name]",
        "subscriptionId": "[subscription().subscriptionId]",
        "uniqueStorage": "[uniqueString(variables('subscriptionId'), variables('serverResourceGroupName'), parameters('location'))]",
-       "storageName": "[tolower(concat('sqlva', variables('uniqueStorage')))]"
+       "storageName": "[tolower(concat('sqlva', variables('uniqueStorage')))]",
+       "uniqueRoleGuid": "[guid(uniqueString(variables('subscriptionId'), variables('serverResourceGroupName'), parameters('serverName'), parameters('location')))]",
```
Comment (bmoore-msft) [inline 101-sql-logical-server/azuredeploy.json]: use:

guid(resourceId('Microsoft.Sql/servers', parameters('severName'))

File: 101-sql-logical-server/azuredeploy.json [hunk 3]
```diff
@@ -109,7 +114,7 @@
                   ],
                   "properties": {
                       "storageContainerPath": "[if(parameters('enableADS'), concat(reference(resourceId('Microsoft.Storage/storageAccounts', variables('storageName')), '2018-07-01').primaryEndpoints.blob, 'vulnerability-assessment'), '')]",
-                      "storageAccountAccessKey": "[if(parameters('enableADS'), listKeys(resourceId('Microsoft.Storage/storageAccounts', variables('storageName')), '2018-02-01').keys[0].value, '')]",
+                      "storageAccountAccessKey": "",
```
Comment (bmoore-msft) [inline 101-sql-logical-server/azuredeploy.json]: remove empty property

File: 101-sql-logical-server/azuredeploy.json [hunk 4]
```diff
@@ -139,6 +144,21 @@
...
+          "type": "Microsoft.Storage/storageAccounts/providers/roleAssignments",
+          "apiVersion": "2018-09-01-preview",
+          "name": "[concat(variables('storageName'), '/Microsoft.Authorization/', variables('uniqueRoleGuid') )]",
+          "dependsOn": [
+              "[resourceId('Microsoft.Sql/servers', parameters('serverName'))]",
+              "[resourceId('Microsoft.Storage/storageAccounts', variables('storageName'))]"
+          ],
+          "properties": {
+              "roleDefinitionId": "[variables('StorageBlobContributor')]",
+              "principalId": "[reference(resourceId('Microsoft.Sql/servers', parameters('serverName')), '2015-05-01-preview' , 'Full').identity.principalId]",
```
Comment (bmoore-msft) [inline 101-sql-logical-server/azuredeploy.json]: update apiVersion or just remove it

File: 101-sql-logical-server/azuredeploy.json [hunk 5]
```diff
@@ -139,6 +144,21 @@
...
+    {
+          "condition": "[parameters('enableADS')]",
+          "type": "Microsoft.Storage/storageAccounts/providers/roleAssignments",
+          "apiVersion": "2018-09-01-preview",
+          "name": "[concat(variables('storageName'), '/Microsoft.Authorization/', variables('uniqueRoleGuid') )]",
+          "dependsOn": [
+              "[resourceId('Microsoft.Sql/servers', parameters('serverName'))]",
+              "[resourceId('Microsoft.Storage/storageAccounts', variables('storageName'))]"
+          ],
+          "properties": {
```
Comment (bmoore-msft) [inline 101-sql-logical-server/azuredeploy.json]: add the principalType property:

 "principalType": "servicePrincipal",

**Pull request closed without merging**